src/core/MOM_open_boundary.F90

```diff
@@ -828,6 +987,66 @@ subroutine initialize_segment_data(G, OBC, PF)
 
 end subroutine initialize_segment_data
 
+subroutine initialize_obc_tides(OBC, tide_ref_date, nodal_ref_date, tide_constituent_str)
+  type(ocean_OBC_type), pointer       :: OBC !< Open boundary control structure
+  integer, dimension(3), intent(in) :: tide_ref_date      !< Reference date (t = 0) for tidal forcing.
+  integer, dimension(3), intent(in) :: nodal_ref_date     !< Date to calculate nodal modulation for.
+  character(len=50), intent(in) :: tide_constituent_str   !< List of tidal constituents to include on boundary.
+  real, dimension(4) :: nodal_shpn                        !< Solar and lunar longitudes for tidal forcing
+  real :: nodal_time                                      !< Model time to calculate nodal modulation for.
+  integer :: c                                            !< Index to tidal constituent.
+
+  allocate(OBC%tide_names(OBC%n_tide_constituents))
+  read(tide_constituent_str, *) OBC%tide_names
+
+  ! Set reference time (t = 0) for boundary tidal forcing.
+  OBC%time_ref = time_type_to_real(set_date(tide_ref_date(1), tide_ref_date(2), tide_ref_date(3)))
+
+  ! Find relevant lunar and solar longitudes at the reference time
+  if (OBC%add_eq_phase) call astro_longitudes_init(OBC%time_ref, OBC%astro_shpn)
+
+  ! If the nodal correction is based on a different time, initialize that.
+  ! Otherwise, it can use N from the time reference.
+  if (OBC%add_nodal_terms) then
+    if (sum(nodal_ref_date) .ne. 0) then
+      ! A reference date was provided for the nodal correction
+      nodal_time = time_type_to_real(set_date(nodal_ref_date(1), nodal_ref_date(2), nodal_ref_date(3)))
+      call astro_longitudes_init(nodal_time, nodal_shpn)
+    elseif (OBC%add_eq_phase) then
+      ! Astronomical longitudes were already calculated for use in equilibrium phases,
+      ! so use nodal longitude from that.
+      nodal_shpn = OBC%astro_shpn
```

Comment:
This is an array syntax copy, which needs to use syntax that is consistent with what is happening here, namely this should be `nodal_shpn(:) = OBC%astro_shpn(:)`.  Also since this is an array of 4 different longitudes, perhaps this would be clearer if it were turned into a type with named elements and comments describing each one.